public/src/components/amounts/AmountsForm.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { useState } from 'react';
+import {
+  Territory,
+  AmountsTests,
+  AmountsTest,
+  CountryOptions,
+  ContributionType,
+} from '../../utils/models';
+
+import { AmountsTestsList } from './AmountsTestsList';
+import { AmountsTestEditor } from './AmountsTestEditor';
+
+import {
+  SupportFrontendSettingsType,
+  fetchSupportFrontendSettings,
+  saveSupportFrontendSettings,
+} from '../../utils/requests';
+import withS3Data, { InnerProps, DataFromServer } from '../../hocs/withS3Data';
+
+import { makeStyles, Theme } from '@material-ui/core/styles';
+
+const useStyles = makeStyles(({ spacing }: Theme) => ({
+  body: {
+    display: 'flex',
+    overflow: 'hidden',
+    flexGrow: 1,
+    width: '100%',
+    height: '100%',
+  },
+  leftCol: {
+    height: '100%',
+    flexShrink: 0,
+    overflowY: 'auto',
+    background: 'white',
+    paddingTop: spacing(6),
+    paddingLeft: spacing(6),
+    paddingRight: spacing(6),
+  },
+  rightCol: {
+    overflowY: 'auto',
+    flexGrow: 1,
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'flex-start',
+    paddingTop: spacing(6),
+    paddingLeft: spacing(6),
+  },
+}));
+
+const AmountsForm: React.FC<InnerProps<AmountsTests>> = ({
+  data: configuredAmounts,
+  setData: setConfiguredAmounts,
+  saveData: saveConfiguredAmounts,
+}: InnerProps<AmountsTests>) => {
+  const classes = useStyles();
+
+  const [selectedTest, setSelectedTest] = useState<AmountsTest | undefined>();
+
+  const onTargetSelected = (target: Territory) => {
+    const currentTest = configuredAmounts.filter(test => test.target === target);
+    if (currentTest.length) {
+      setSelectedTest(currentTest[0]);
```

Comment:
you can use `find` instead of `filter` if you only expect 0 or 1 values

Reply:
Updated as suggested